llpc/patch/llpcPatchBufferOp.cpp

```diff
@@ -1564,512 +1585,279 @@ Value* PatchBufferOp::ReplaceLoad(
         Value* const pNewBaseIndex = m_pBuilder->CreateSelect(pInBound, pBaseIndex, m_pBuilder->getInt32(0));
 
         // Add on the index to the address.
-        Value* pLoadPointer = m_pBuilder->CreateGEP(pBaseAddr, pNewBaseIndex);
-
-        pLoadPointer = m_pBuilder->CreateBitCast(pLoadPointer, pLoadType->getPointerTo(ADDR_SPACE_GLOBAL));
+        Value* pPointer = m_pBuilder->CreateGEP(pBaseAddr, pNewBaseIndex);
 
-        LoadInst* const pNewLoad = m_pBuilder->CreateLoad(pLoadPointer);
-        pNewLoad->setVolatile(pLoadInst->isVolatile());
-        pNewLoad->setAlignment(MaybeAlign(pLoadInst->getAlignment()));
-        pNewLoad->setOrdering(pLoadInst->getOrdering());
-        pNewLoad->setSyncScopeID(pLoadInst->getSyncScopeID());
-        CopyMetadata(pNewLoad, pLoadInst);
+        pPointer = m_pBuilder->CreateBitCast(pPointer, pType->getPointerTo(ADDR_SPACE_GLOBAL));
 
-        if (isInvariant)
+        if (isLoad)
         {
-            pNewLoad->setMetadata(LLVMContext::MD_invariant_load, MDNode::get(*m_pContext, None));
+            LoadInst* const pNewLoad = m_pBuilder->CreateLoad(pPointer);
+            pNewLoad->setVolatile(pLoadInst->isVolatile());
```

Comment:
`volatile`, ordering, etc. can checking can be hoisted to top of the function; it's already done for alignment and ordering.

Reply:
Good catch. I changed it to use the variables from above.
`isVolatile` is only used in this part, so I didn’t want to hoist it.